notify/notify.go

```diff
@@ -645,17 +658,17 @@ func (n *DedupStage) Exec(ctx context.Context, _ log.Logger, alerts ...*types.Al
 // RetryStage notifies via passed integration with exponential backoff until it
 // succeeds. It aborts if the context is canceled or timed out.
 type RetryStage struct {
-	integration Integration
-	groupName   string
-	metrics     *Metrics
+	integration  *Integration
+	receiverName string
+	metrics      *Metrics
 }
 
 // NewRetryStage returns a new instance of a RetryStage.
-func NewRetryStage(i Integration, groupName string, metrics *Metrics) *RetryStage {
+func NewRetryStage(i *Integration, receiverName string, metrics *Metrics) *RetryStage {
```

Comment:
I think I would prefer to keep groupName even though it can be confusing.
1. It will reduce the number of changes, which may help us push it upsteram